src/frontend/test_runner/tests/testdata/join.yaml

```diff
@@ -276,3 +278,141 @@
     select * from a join lateral (select * from b where a1 = b1);
   binder_error: 'Feature is not yet implemented: lateral subqueries are not yet supported,
     Tracking issue: https://github.com/singularity-data/risingwave/issues/3815'
+- sql: |
+    /* Ensure that natural joins bind the correct columns */
+    create table a(x int);
+    create table b(x int);
+    create table c(y int);
+    select * from a natural join b natural join c;
+  logical_plan: |
+    LogicalProject { exprs: [c.y, a.x] }
+      LogicalJoin { type: Inner, on: true }
+        LogicalJoin { type: Inner, on: (a.x = b.x) }
+          LogicalScan { table: a, columns: [_row_id, x] }
+          LogicalScan { table: b, columns: [_row_id, x] }
+        LogicalScan { table: c, columns: [_row_id, y] }
+- sql: |
+    /* Ensure that natural joins can disambiguate columns */
+    create table a(x int);
+    create table b(x int);
+    select x, a.x, b.x from a natural join b;
+  logical_plan: |
+    LogicalProject { exprs: [a.x, a.x, b.x] }
+      LogicalJoin { type: Inner, on: (a.x = b.x) }
+        LogicalScan { table: a, columns: [_row_id, x] }
+        LogicalScan { table: b, columns: [_row_id, x] }
+- sql: |
+    /* Ensure that natural joins bind the correct columns */
+    create table a(x int);
+    create table b(x int);
+    create table c(y int);
+    select * from a natural join b natural join c;
+  logical_plan: |
+    LogicalProject { exprs: [c.y, a.x] }
+      LogicalJoin { type: Inner, on: true }
+        LogicalJoin { type: Inner, on: (a.x = b.x) }
+          LogicalScan { table: a, columns: [_row_id, x] }
+          LogicalScan { table: b, columns: [_row_id, x] }
+        LogicalScan { table: c, columns: [_row_id, y] }
+- sql: |
+    /* Ensure that natural joins can disambiguate columns */
+    create table a(x int);
+    create table b(x int);
+    select x, a.x, b.x from a natural join b;
+  logical_plan: |
+    LogicalProject { exprs: [a.x, a.x, b.x] }
+      LogicalJoin { type: Inner, on: (a.x = b.x) }
+        LogicalScan { table: a, columns: [_row_id, x] }
+        LogicalScan { table: b, columns: [_row_id, x] }
+- sql: |
+    /* Ensure that natural joins bind the correct columns */
+    create table a(x int);
+    create table b(x int);
+    create table c(y int);
+    select * from a natural join b natural join c;
+  logical_plan: |
+    LogicalProject { exprs: [c.y, a.x] }
```

Comment:
Is the ordering of expressions correct here? Seems like we may need to ensure a correct ordering?

Reply:
> It may be as simple as putting all the ambiguous references first (to comply with postgres)

I believe this is the correct solution.